nbclient/client.py

```diff
@@ -344,13 +354,13 @@ async def _async_cleanup_kernel(self):
         finally:
             # Remove any state left over even if we failed to stop the kernel
             await ensure_async(self.km.cleanup())
-            if getattr(self, "kc"):
+            if self.kc is not None:
```

Comment:
I believe this one actually needs to be `getattr(self, "kc")` or `getattr(self, "kc") is not None` because the KernelGateway (and possibly other?) code paths sometimes don't assign any kc to the object by the time it reaches this call sometimes. I see you were testing this with asserts within the functions in places, but I can't quite recall the exact circumstances that cause the issue originally.

Reply:
So I changed it because otherwise we get `error: Item "None" of "Optional[Any]" has no attribute "stop_channels"` with the following line. I think it should be:
```python
if getattr(self, "kc") and self.kc is not None:
```

Reply:
sounds good to me